src/nix.rs

```diff
@@ -251,7 +251,7 @@ pub fn open(mut new_file: SharedMem) -> Result<SharedMem> {
 pub fn create(mut new_file: SharedMem, lock_type: LockType) -> Result<SharedMem> {
     #[cfg(target_os="macos")]
     let max_path_len = 30;
-    #[cfg(target_os="linux")]
+    #[cfg(any(target_os="freebsd", target_os="linux"))]
     let max_path_len = 255;
```

Comment:
I am not sure how to verify that this limit is correct for FreeBSD. If you can point to where the value is defined in Linux I'm happy to double check.

Reply:
The limit is in fact larger on FreeBSD (from reading the source sys/kern/uipc_shm.c, 1023 bytes minus the length of the path to the jail root (if any), for the whole pathname including initial slash), but I don't think using that extra space is particularly useful. With 255 it is portable and safe except in pathological cases with jails.